### PR TITLE
TWC2 add whatsapp deep link

### PR DESCRIPTION
**Pull request by e056**
Fixes https://github.com/bettersg/call-home/issues/131

https://github.com/bettersg/call-home/issues/131#issuecomment-1445155469
image

### CODE REVIEW

**Comment by e056** [comment]
Sorry let me resolve the merge conflicts first

**Comment by glencbz** [comment]
It looks like this PR is based on an old version of your previous PR (e.g. https://github.com/bettersg/call-home/commit/0b10757d2dfafbf0d03b0469146a314574501b91) instead of the latest version of `master`, since your PR was rewritten quite a lot before being merged to master, (https://github.com/bettersg/call-home/pull/103/files).

I think the easiest way to get the changes in the right shape is to rebase your work onto the latest `master`. This will let us drop the merge commit and to pretend as if we had tried to write the code on `master` in the first place (i.e. as if we had 'based' the work on `master` instead of our local WIP, hence 'changing the base'). From inspecting this log
image

I think it might not be too difficult. The first commit 0b10757d (Add Navbar footer component, 2022-12-18) contains the entirety of your old PR, so your new changes are just the second and third commits. If we are rebasing, we can also drop the last merge commit c123c8b3 (Merge remote-tracking branch 'origin' into twc2-add-whatsapp-deep-link, 2023-03-02), since that was supposed to 'update' our branch to be up to date with `master`, but we don't need that if we are already basing our work on the new `master`.

Here's one way to do it:

```bash
git checkout master
# update our master branch to make sure it's up to date
git pull
# or whatever the branch is called locally
git checkout twc2-add-whatsapp-deep-link
# create a backup branch before we do anything!! This will make it easy to recover if we screw up
git branch BACKUP-twc2-add-whatsapp-deep-link 
git rebase master # rebase our changes onto master
```

This rebase will go through our commits one by one and try to apply them onto `master`. This process will have a whole bunch of merge conflicts, which is expected, since we originally made the changes on a very different branch from the one we want to apply it to. Nevertheless, it should be manageable :)

- We can skip the first commit, because it's just creating the navbar, but the navbar changes are already in master. `git rebase --skip`.
- We can also skip the second commit because it's just applying changes to bring your old PR up to date with the merged version (If I understand it correctly). The conflicted changes are just things that got rewritten. `git rebase --skip`.
- The third commit requires some merge commit resolution. I suggest getting a quick tutorial on merge conflict resolution (maybe [this](https://www.youtube.com/watch?v=xNVM5UxlFSA&ab_channel=Ihatetomatoes)?) before starting. Most of the conflicted files don't actually contain relevant changes. For those files, you can do `git restore --ours <name-of-file> && git add <name-of-file>` to drop the changes from your commit. 

Especially if you aren't familiar with the process, I suggest doing this incrementally, i.e. don't feel the need to get everything right on the first try. For example, on the first pass, I might get rid of 10% of the files that I didn't want to change, then `git push --force` to GitHub and inspect the result on https://github.com/bettersg/call-home/pull/137/files. If that looks good, I can repeat the process and remove 50% of the files, etc etc.

Also, let me know if I can help by getting on a call/getting someone to provide a second pair of eyes.

**Comment by glencbz** [comment]
Actually, if you don't want to go through the trouble of rebasing, it might be even easier to just redo the PR :) something like:

```bash
# Copy out the files that were changed
mkdir /tmp/my-call-home-work
cp the-files-i-changed /tmp/my-call-home-work
# Let's check out origin/master because that's the version on GitHub. Our own master branch might have unexpected changes
git checkout origin/master
# Create and checkout a new branch
git checkout -b twc2-cta-button
# Restore the changes
cp /tmp/my-call-home-work/bla/bla .
```

File: apps/super-app/src/routes/SupportDetailWip.tsx
Comment (stevanussatria) [inline apps/super-app/src/routes/SupportDetailWip.tsx:120]: @e056 this and the entire primary button is the only thing that's new right? Can we make it into a component that takes in props instead?

The props will be for:
- The link to redirect to
- The icon to be used (not all will be WA)
- The text for the button label

**Review by stevanussatria — CHANGES_REQUESTED** [review]

**Comment by stevanussatria** [comment]
@e056 i went through the PR. If I'm not wrong most of the changes are due to conflicts/outdated code, and only the primary button is the new code change?

If so might be easier to just redo the PR fresh from the latest `master` branch.

Let's also componentize the button 😊

**Comment by e056** [comment]
Creating new PR instead

**Pull request closed without merging**
